test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -381,5 +381,66 @@ public void GetsKnownVersion(string input, char endChar, bool expectedResult, st
 
             _pool.Return(block);
         }
+
+        [Theory]
+        [InlineData("HTTP/1.0\r", "HTTP/1.0")]
+        [InlineData("HTTP/1.1\r", "HTTP/1.1")]
+        public void KnownVersionsAreInterned(string input, string expected)
+        {
+            TestKnownStringsInterning(input, expected,
+                (iter, expect) => {
+                    string output;
+                    iter.GetKnownVersion(out output);
+                    Assert.True(output == expect);
```

Comment:
Testing different references, but making sure its returning the right thing (not null)